src/shared/gpt.c

```diff
@@ -14,87 +14,55 @@
-        { GPT_USR_IA64_VERITY_SIG,         "usr-ia64-verity-sig"         },
-        { GPT_USR_LOONGARCH64,             "usr-loongarch64"             },
-        { GPT_USR_LOONGARCH64_VERITY,      "usr-loongarch64-verity"      },
-        { GPT_USR_LOONGARCH64_VERITY_SIG,  "usr-loongarch64-verity-sig"  },
-        { GPT_USR_RISCV32,                 "usr-riscv32"                 },
-        { GPT_USR_RISCV32_VERITY,          "usr-riscv32-verity"          },
-        { GPT_USR_RISCV32_VERITY_SIG,      "usr-riscv32-verity-sig"      },
-        { GPT_USR_RISCV64,                 "usr-riscv64"                 },
-        { GPT_USR_RISCV64_VERITY,          "usr-riscv64-verity"          },
-        { GPT_USR_RISCV64_VERITY_SIG,      "usr-riscv64-verity-sig"      },
-        { GPT_USR_X86,                     "usr-x86"                     },
-        { GPT_USR_X86_64,                  "usr-x86-64"                  },
-        { GPT_USR_X86_64_VERITY,           "usr-x86-64-verity"           },
-        { GPT_USR_X86_64_VERITY_SIG,       "usr-x86-64-verity-sig"       },
-        { GPT_USR_X86_VERITY,              "usr-x86-verity"              },
-        { GPT_USR_X86_VERITY_SIG,          "usr-x86-verity-sig"          },
-#ifdef GPT_USR_NATIVE
-        { GPT_USR_NATIVE,                  "usr"                         },
-        { GPT_USR_NATIVE_VERITY,           "usr-verity"                  },
-        { GPT_USR_NATIVE_VERITY_SIG,       "usr-verity-sig"              },
+        { GPT_ROOT_NATIVE,            "root"                       },
+        { GPT_ROOT_NATIVE_VERITY,     "root-verity"                },
+        { GPT_ROOT_NATIVE_VERITY_SIG, "root-verity-sig"            },
+        { GPT_USR_NATIVE,             "usr"                        },
+        { GPT_USR_NATIVE_VERITY,      "usr-verity"                 },
+        { GPT_USR_NATIVE_VERITY_SIG,  "usr-verity-sig"             },
 #endif
-#ifdef GPT_USR_SECONDARY
-        { GPT_USR_SECONDARY,               "usr-secondary"               },
-        { GPT_USR_SECONDARY_VERITY,        "usr-secondary-verity"        },
-        { GPT_USR_SECONDARY_VERITY_SIG,    "usr-secondary-verity-sig"    },
-#endif
-        { GPT_ESP,                         "esp"                         },
-        { GPT_XBOOTLDR,                    "xbootldr"                    },
-        { GPT_SWAP,                        "swap"                        },
-        { GPT_HOME,                        "home"                        },
-        { GPT_SRV,                         "srv"                         },
-        { GPT_VAR,                         "var"                         },
-        { GPT_TMP,                         "tmp"                         },
-        { GPT_USER_HOME,                   "user-home"                   },
-        { GPT_LINUX_GENERIC,               "linux-generic"               },
+        _GPT_ARCH_SEXTET(SECONDARY,   "secondary"),
```

Comment:
hmm this should require #ifdeffery, no? i mean only some archs have a secondary arch?

Reply:
Oh, indeed. I don't know why I dropped the the `#ifdef`.